### DISCUSSION TITLE
USR-TCP232-304

### DISCUSSION THREAD

**Discussion by Chrisbrown01942** [nathanmarlor/foxess_em#322] [Q&A]
Hello, is there an issue with using the USR-TCP232-304 with the FoxESS_Energy_Management? I've recently swapped out my USR-W610 for a USR-TCP232-304 which works with the FoxESS_Modbus intergration no problems. However, when configuring the FoxESS_Energy_Management integration with the IP address it fails. For now I'm using the cloud based option, but this doesn't allow the custom charge profile. A great feature of the Energy Management integration.

Many thanks, Chris

**Comment by nathanmarlor** [comment]
Hi Chris, there shouldn't be any issues - can you post some logs please and I'll take a look at what's happening 🙂

**Reply by Chrisbrown01942** [reply]
Hi, thanks for getting back to me. Log below, that appears when trying to config with the IP address.

This error originated from a custom integration.

Logger: custom_components.foxess_em.config_flow
Source: custom_components/foxess_em/config_flow.py:250
Integration: FoxESS - Energy Management (documentation, issues)
First occurred: 09:33:59 (1 occurrences)
Last logged: 09:33:59

ModbusIOException()

**Reply by nathanmarlor** [reply]
Thanks - are there any other logs/exceptions around the error? Can you detail how you have the adapter setup on the modbus please? TCP etc?

**Comment by Chrisbrown01942** [comment]
Hi, I've set-up the device as per the instructions on the wiki (I've attatched an image). Also, I switched on debugging then tried to config again. The I/O error is at the bottom of the text. I've also cleared the logs & run the config again. The only error in the log at the moment is the Modbus I/O error, so nothing else around the error?

Regards, Chris
[home-assistant_foxess_em_2023-07-21T09-31-35.766Z.log](https://github.com/nathanmarlor/foxess_em/files/12122272/home-assistant_foxess_em
![Screenshot 2023-07-21 103500](https://github.com/nathanmarlor/foxess_em/assets/118054494/93b9ae2c-6df0-470e-b67d-eb9c0f0ad50c)
_2023-07-21T09-31-35.766Z.log)

**Reply by Chrisbrown01942** [reply]
The debug file
[home-assistant_foxess_em_2023-07-21T09-31-35.766Z.log](https://github.com/nathanmarlor/foxess_em/files/12122338/home-assistant_foxess_em_2023-07-21T09-31-35.766Z.log)

**Reply by nathanmarlor** [reply]
Thanks. The logs look like it's trying to connect to a 192.168.1.191 address, however the adapter is setup on a 192.168.0.x?

**Comment by Chrisbrown01942** [comment]
Hi, apologies I'm a little confused. I set up the 302 & foxess_modbus integration on 192.168.1.191 which works ok. Re the server address in the image above, the IP 192.168.0.201 was already set (I didn't input anything). I wasn't sure if I should change anything with that, so left as is, & the foxess_modbus integration is functioning properly, including writing to the registers (changing charge amps etc.).  Should everything on the 302 be pointing to 192.168.1.191. For example, on the 302 the static IP is 192.168.1.191 set on the router. Should the server address also point to 192.168.1.191. Then setting up the Energy Management, should I set it to 192.168.1.191?

Regards, Chris

**Comment by Chrisbrown01942** [comment]
Hi, I ran the modpoll exe to check the device is working. In the config for the EM I'm using the IP address of the 302 (192.168.1.191) just don't understand why it fails to connect.
![Screenshot 2023-07-21 173735](https://github.com/nathanmarlor/foxess_em/assets/118054494/9520377f-f24f-419f-86c8-a0b6e5f00feb)

**Comment by canton7** [comment]
The USR-TCP232-304 is a TCP to serial adapter, which does not know anything about modbus / ModbusTCP. foxess_modbus supports this sort of adapter, but foxess_em does not.

You will probably hit problems if you try and have two integrations using the USR-TCP232-304 at the same time: if two things try and send serial commands to the USR-TCP232-304 at the same time it probably won't be able to send the right response to the right integration.

**Comment by Chrisbrown01942** [comment]
Hi, cool I understand. Once again thanks for getting back to me. The USR-W610 worked with both integrations (foxess modbus & foxess EM), but was slow in TCP mode polling every 15 seconds. The UDP option is faster with the foxess modbus, but like the 302 doesn't work with foxess modbus & foxessEM. Before I go & purchase a 402s, have you seen examples of the 402s working with both the foxess modbus & foxessEM?

Regards, Chris
